src/Conversion/ONNXToTorch/NN/LeakyReluOp.cpp

```diff
@@ -55,46 +51,31 @@ using namespace mlir::torch::Torch;
  *
  */
 
-class ONNXLeakyReluOpToTorchLowering : public ConversionPattern {
+class ONNXLeakyReluOpToTorchLowering : public OpConversionPattern<ONNXLeakyReluOp> {
 public:
-  ONNXLeakyReluOpToTorchLowering(TypeConverter &typeConverter, MLIRContext *ctx)
-      : ConversionPattern(
-            typeConverter, mlir::ONNXLeakyReluOp::getOperationName(), 1, ctx) {}
-
-  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
-      ConversionPatternRewriter &rewriter) const final {
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult matchAndRewrite(ONNXLeakyReluOp op, OpAdaptor adaptor,
+        ConversionPatternRewriter &rewriter) const override {
 
-    Location loc = op->getLoc();
+    Location loc = op.getLoc();
     mlir::MLIRContext *context = op->getContext();
-    ONNXLeakyReluOp op1 = llvm::dyn_cast<ONNXLeakyReluOp>(op);
-    ONNXLeakyReluOpAdaptor adapter(op1);
-
-    Value x = op1.X();
 
-    auto alpha = adapter.alphaAttr(); // mlir::FloatAttr
+    auto x = adaptor.X();
+    auto alpha = adaptor.alphaAttr(); // mlir::FloatAttr
     auto negSlope = alpha.getValue(); // APSFloat
     auto negSlopeFloatValue = FloatAttr::get(
-        mlir::FloatType::getF64(op->getContext()), negSlope.convertToFloat());
+        mlir::FloatType::getF64(op.getContext()), negSlope.convertToFloat());
     Value negSlopeConstFloat =
         rewriter.create<ConstantFloatOp>(loc, negSlopeFloatValue);
 
-    TensorType xTensorType = x.getType().cast<TensorType>();
-    TensorType opTensorType = op->getResult(0).getType().cast<TensorType>();
-
-    auto xType = Torch::ValueTensorType::get(
-        context, xTensorType.getShape(), xTensorType.getElementType());
-    auto xTorchTensor =
-        rewriter.create<torch::TorchConversion::FromBuiltinTensorOp>(
-            loc, xType, x);
-    auto resultType = Torch::ValueTensorType::get(op1.getContext(),
+    TensorType opTensorType = op.getResult().getType().cast<TensorType>();
+    auto resultType = Torch::ValueTensorType::get(op->getContext(),
```

Comment:
This is not very idiomatic. The TypeConverter is already attached to this pattern `getTypeConverter()` and therefore should be used to perform this task. I would expect to find `getTypeConverter()->convertType(op.getResult().getType())` here.